pennylane/estimator/ops/op_math/symbolic.py

```diff
@@ -574,16 +540,7 @@ def resource_decomp(
             if key in base_params and base_params[key] is None
         )
 
-        if pow_z == 0:
-            return [GateCount(resource_rep(qre.Identity))]
-
-        if pow_z == 1:
-            return [GateCount(base_cmpr_op)]
```

Comment:
Should we leave these two if statements here? The idea being if we don't then all other custom pow_resource_decomp will have to have them manually.

Reply:
For `estimate(Pow(base(), 0))`, I would expect the resources to match exactly as described in the base's `pow_resource_decomp`. I would be a bit surprised to get something that's _not_ described in the `pow_resource_decomp` method.

For example, if we did `estimate(Pow(CRX(), 0))`, what should the expected result be? I would read `CRX`'s `pow_resource_decomp` and see:

```
        return [GateCount(cls.resource_rep(precision))]
```
and expect to get [CRX] for my estimated result. But our current pipeline would return an Identity gate, which, while correct, can lead to a frustrating experience when people have to go find where this Identity gate came from. 

My goal with this refactor is to make `pow_resource_decomp` the single source of truth.